src/main/java/de/taimos/pipeline/aws/WaitDeployStep.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * -
+ * #%L
+ * Pipeline: AWS Steps
+ * %%
+ * Copyright (C) 2017 Taimos GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package de.taimos.pipeline.aws;
+
+import com.amazonaws.services.codedeploy.AmazonCodeDeployClient;
+import com.amazonaws.services.codedeploy.model.GetDeploymentRequest;
+import com.amazonaws.services.codedeploy.model.GetDeploymentResult;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousNonBlockingStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.inject.Inject;
+
+/**
+ * @author Giovanni Gargiulo
+ */
+public class WaitDeployStep extends AbstractStepImpl {
+
+    /**
+     * The DeploymentId to monitor. Example: d-3GR0HQLDN
+     */
+    private final String deploymentId;
+
+    /**
+     * The max amount of time, before failing be build.
+     * A negative number will mean indefinite wait
+     */
+    private final Long maxWait;
+
+    @DataBoundConstructor
+    public WaitDeployStep(String deploymentId, Long maxWait) {
+        this.deploymentId = deploymentId;
+        this.maxWait = maxWait;
+    }
+
+    public String getDeploymentId() {
+        return deploymentId;
+    }
+
+    @Extension
+    public static class DescriptorImpl extends AbstractStepDescriptorImpl {
+
+        public DescriptorImpl() {
+            super(Execution.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return "waitDeploymentCompletion";
```

Comment:
maybe "awaitDeploymentCompletion" would be better or "waitForDeploymentCompletion"